File: v1/Demo/demo_set_current_pid.py
import sys, os
sys.path.append(os.path.join(os.path.dirname(__file__), "../"))
from Hand_Sdk.HandSocket import *
from Hand_Sdk.Handapi import *
import time, math


def main():
    ip_left = "192.168.137.39" # lefthand
    # ip_right = "192.168.137.19" # righthand
    api = API(ip_left)
    api.calibration()
    time.sleep(1)

    # set
    id = 1
    _pid = [4.0, 2.0, 0.0]
    if api.set_current_pid(id, _pid) == FunctionResult.FAIL:
        logger.print_trace_error("set_current_pid failed")
    time.sleep(0.01)

    # verify
    current_pid = api.get_current_pid()
    if len(current_pid) != 18:
        logger.print_trace_error("get current_pid failed")
    else:
        logger.print_trace("get current_pid success", current_pid)



if __name__ == "__main__":
    main()

File: v1/Demo/source/demo_set_current_pid.cpp
#include "../include/Handapi.h"

using namespace HandProtocol;

int main()
{
    Handapi api;
    api.init(LEFT_IP);
    api.calibration();
        
    // set
    uint8_t id = 1;
    vector<float> _pid {4.0, 2.0, 0.0};
    if (api.set_current_pid(id, _pid) == ResultCode::FAIL)
    {
        Logger::get_instance()->print_trace_error("set set_current_pid failed\n");
    }
    OS::get_instance()->delay_s(1);

    // verify
    vector<float> current_pid = api.get_current_pid();
    if (current_pid.size() != 18)
    {
        Logger::get_instance()->print_trace_error("get current_pid failed\n");
        return ResultCode::FAIL;
    }
    
    std::cout<<"get current_pid success : ";
    for (int i = 0; i < current_pid.size(); ++i) {
        std::cout << current_pid[i] << " ";
    }
    std::cout << std::endl;

    return 0;

}